test/t8_cmesh/t8_gtest_cmesh_partition.cxx

```diff
@@ -0,0 +1,154 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <gtest/gtest.h>
+#include <t8_cmesh.h>
+#include <t8_schemes/t8_default/t8_default_cxx.hxx>
+#include "t8_cmesh/t8_cmesh_trees.h"
+#include "t8_cmesh/t8_cmesh_partition.h"
+#include <t8_cmesh/t8_cmesh_testcases.h>
+
+/* We create a cmesh, partition it and repartition it several times.
+ * At the end we result in the same partition as at the beginning and we
+ * compare this cmesh with the initial one. If they are equal the test is
+ * passed.
+ *
+ * TODO: - Currently the test is not passed. This is probably because the
+ *         cmesh_is_equal function is too strict and does not allow, for example,
+ *         the order of the ghosts to change.
+ *         We should implement a lighter version of cmesh_is_equal in order
+ *         to account for this.
+ * 
+ *       - when this test works for all cmeshes remove if statement in 
+ *         classes and tests 
+ */
+
+/* *INDENT-OFF* */
+class t8_cmesh_partition_class : public testing::TestWithParam<int>{
+protected:
+  void SetUp() override {
+    cmesh_id = GetParam();
+
+    if (!(cmesh_id != 89 && (cmesh_id < 237 || cmesh_id > 256))) {
+      GTEST_SKIP();
+    }
+    cmesh_original = t8_test_create_cmesh (cmesh_id);
+  }
+
+  int                 cmesh_id;
+  t8_cmesh_t          cmesh_original;
+};
+/* *INDENT-ON* */
+
+static void
+test_cmesh_commited (t8_cmesh_t cmesh)
+{
+  ASSERT_TRUE (t8_cmesh_is_committed (cmesh)) << "Cmesh commit failed.";
+  ASSERT_TRUE (t8_cmesh_trees_is_face_consistend (cmesh, cmesh->trees)) <<
+    "Cmesh face consistency failed.";
+}
+
+TEST_P (t8_cmesh_partition_class, test_cmesh_partition_concentrate)
+{
+  /* This if statement is necessary to make the test work by avoiding specific cmeshes which do not work yet for this test.
```

Comment:
This comment seems to be wrong. The following `#ifdef` is used to minimize the number of tests in different configurations of t8code. 
I think you want to move it to line 50.
Can you also check, if the condition is still necessary?